cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -179,6 +181,13 @@ data SetupScriptOptions = SetupScriptOptions {
     -- style with no dependencies.
     useDependenciesExclusive :: Bool,
 
+    -- | Don't add a default dependency on Cabal even if
+    -- 'useDependenciesExclusive' is @False@. Assume that it is provided via
+    -- 'useDependencies'. This is turned on (in conjunction with
+    -- useDependenciesExclusive = False) when a default setup dependencies
+    -- stanza is added by cabal-install itself. See #3199.
+    noDefaultCabalDep :: Bool,
```

Comment:
So, is useDependenciesExclusive == True, noDefaultCabalDep == True and impossible state?